test/accumulate.jl

```diff
@@ -0,0 +1,12 @@
+using CuArrays
+
+v = [1, 2, 3]
+v2 = CuArray(v)
+
+@testset "accumulate! tests"
```

Comment:
maybe also directly test `accumulate`/`accumulate!`
we also typically use `testf` to compare against the `Array` output instead of hardcoding results.